src/components/PopupWithForm.js

```diff
@@ -18,17 +19,40 @@ export default class PopupWithForm extends Popup {
     return formValues;
   }
 
+  _setInputValues(data) {
+    this._inputList.forEach((input) => {
+      input.value = data[input.name] || "";
+    });
+  }
+
   _setEventListeners() {
-    this._popupForm.addEventListener("submit", (event) => {
+    this._popupForm.addEventListener("submit", async (event) => {
       event.preventDefault();
       const data = this._getInputValues();
-      this._handleFormSubmission(data);
-      this.reset();
+      console.log("Form data:", data);
+      this.renderLoading(true);
+      try {
+        await this._handleFormSubmission(data);
+        setTimeout(() => {
+          this.close();
+        }, 300);
+      } catch (error) {
+        console.error("Form submission failed", error);
+      } finally {
+        this.renderLoading(false);
+      }
     });
   }
 
+  renderLoading(isLoading) {
+    if (this._submitButton) {
+      this._submitButton.textContent = isLoading
+        ? this._defaultButtonText
+        : "Saving...";
```

Comment:
**Could be improved**


You can change `"Saving..."` as well if needed

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```